core/system/src/unvic.c

```diff
@@ -68,7 +70,7 @@ void unvic_acl_add(uint8_t box_id, void *function, uint32_t irqn)
     uv = &g_unvic_vector[irqn];
 
     /* check if IRQ entry is populated */
-    if(uv->id || uv->hdlr)
+    if(uv->id != UNVIC_VECTOR_UNUSED)
     {
         HALT_ERROR(PERMISSION_DENIED,
```

Comment:
Instead of halting, which shuts down the entire system even on release builds, consider debug asserting so we can catch programmer mistakes, but silently not fulfill the request in release builds. This will allow us to write tests that make sure we handle interrupt ownership contention. Note that this will require changes to all functions that call `unvic_acl_check`, as the function can now return in the case where the IRQ is registered, but not to the caller. All functions using `unvic_acl_check` will have to deal with: IRQ unclaimed, IRQ claimed by current box, IRQ claimed by other box.